docs/proposals/PROPOSAL_D.md

```diff
@@ -0,0 +1,413 @@
+# Proposal D - Namespaced and Graph-based Compatibility Metadata Schema
+
+> Also maintained by Compatibility Communities
+
+[Proposal C](https://github.com/opencontainers/wg-image-compatibility/pull/8) defines an example compatibility specification that includes namespaced metadata for a compatibility interest group. Missing from that proposal is a way to represent relationships between terms for within a namespace, which would be ideal for representing the compatibility metadata in a graph. Thus, here we want to describe the higher level compatibility definition that defines the allowed terms and relationships for a specific compatibility interest group metadata space, and use the [JSON Graph format](https://jsongraphformat.info/) to map these attributes into a graph space. In summary, there are three things here:
+
+1. A compatibility interest group that defines a known set of metadata attributes and relationships between them (the proposal D, here). There would be one copy of this, in a standard location, one source of truth (versioned).
+2. An example manifestation of the compatibility interest group metadata, specific to one container image to describe it's compatibility ([Proposal C](https://github.com/opencontainers/wg-image-compatibility/pull/8) was a simple, first shot at this, and the format is updated here). These would be paired alongside images in a registry.
+3. jsonschema documents that can be used to validate both of those (for tools to check that the format is valid before using them).
+
+## Why do we want a graph?
+
+There are two use cases I'm currently considering for these compatibility metadata:
+
+- **image selection**: is the primary use case. While at face value a graph might not seem valuable, because an algorithm could very simply (in a brute force manner) match a set of attributes to a desired set. However, given that we have a graph, we can optimize this process by pruning off entire subtrees (that are impossible, for example) before we even consider more detailed metadata. This would hugely be beneficial for use cases we've discussed of needing to perform image selection at a higher level (e.g., asking a question to a registry).
```

Comment:
I think anything more complex than proposed by @sudo-bmitch in the proposal A can be problematic for container runtimes. As I understand, the container runtimes would have to pull an artifact to improve the image selection, is that correct?

Reply:
Agree - I don't know how to deal with this. I suggested in C that we have two contexts for this metadata (alongside platform and then artifact) and I still think that has feet, although I don't think @sudo-bmitch likes the idea. :laughing: